src/test/java/io/jenkins/plugins/analysis/warnings/DockerAgentITest.java

```diff
@@ -0,0 +1,123 @@
+package io.jenkins.plugins.analysis.warnings;
+
+import java.io.ByteArrayOutputStream;
+import java.io.File;
+import java.io.IOException;
+
+import org.junit.BeforeClass;
+import org.junit.Rule;
+import org.junit.Test;
+
+import org.jenkinsci.plugins.workflow.cps.CpsFlowDefinition;
+import org.jenkinsci.plugins.workflow.job.WorkflowJob;
+import org.jenkinsci.test.acceptance.docker.DockerContainer;
+import org.jenkinsci.test.acceptance.docker.DockerRule;
+import org.jenkinsci.test.acceptance.docker.fixtures.JavaContainer;
+import hudson.Launcher;
+import hudson.model.Run;
+import hudson.model.Slave;
+import hudson.plugins.sshslaves.SSHLauncher;
+import hudson.remoting.TeeOutputStream;
+import hudson.slaves.DumbSlave;
+import hudson.util.StreamTaskListener;
+import hudson.util.VersionNumber;
+
+import io.jenkins.plugins.analysis.core.model.AnalysisResult;
+import io.jenkins.plugins.analysis.core.model.ResultAction;
+import io.jenkins.plugins.analysis.core.testutil.IntegrationTestWithJenkinsPerTest;
+
+import static io.jenkins.plugins.analysis.core.model.AnalysisResultAssert.*;
+import static org.hamcrest.Matchers.*;
+import static org.junit.Assume.*;
+
+/**
+ * Integration tests for builds running on a docker agent.
+ *
+ * @author Fabian Janker
+ * @author Andreas Pabst
+ */
+public class DockerAgentITest extends IntegrationTestWithJenkinsPerTest {
+
+    /**
+     * Rule for a docker container with java/maven.
+     */
+    @Rule
+    public DockerRule<JavaContainer> dockerJava = new DockerRule<>(JavaContainer.class);
+
+    /**
+     * Rule for a docker container with gcc/make.
+     */
+    @Rule
+    public DockerRule<GccContainer> dockerGcc = new DockerRule<>(GccContainer.class);
+
+    /**
+     * Assume that this test is run on a unix system and docker is available.
+     *
+     * @throws Exception
+     *         if the version check fails
+     */
+    @BeforeClass
+    public static void unixAndDocker() throws Exception {
+        assumeTrue("This test is only for Unix", File.pathSeparatorChar == ':');
+        ByteArrayOutputStream baos = new ByteArrayOutputStream();
+        assumeThat("`docker version` could be run", new Launcher.LocalLauncher(StreamTaskListener.fromStderr()).launch()
+                .cmds("docker", "version", "--format", "{{.Client.Version}}")
+                .stdout(new TeeOutputStream(baos, System.err))
+                .stderr(System.err)
+                .join(), is(0));
+        assumeThat("Docker must be at least 1.13.0 for this test (uses --init)",
```

Comment:
Please avoid using Harmcrest matchers otherwise the Architecture Tests will fail:
Example:
```
assumeFalse("Docker must be at least 1.13.0 for this test (uses --init)",
                new VersionNumber(baos.toString().trim()).isOlderThan(new VersionNumber("1.13.0")));
```